src/ui/components/Location/component.ts

```diff
@@ -169,33 +159,66 @@ export default class LocationComponent extends Component {
   }
 
   async loadFromAPI(locationSignature: RecordSignature) {
-    let { store } = this.args;
+    let { store, remoteStore } = this.args;
     try {
-      let location: Location = await store.query((q) =>
+      // Fetch data from the API
+      let transform = await remoteStore.pull((q) =>
+        q.findRecord(locationSignature)
+      );
+      // Sync it with the store
+      store.sync(transform);
+      this.location = await store.cache.query((q) =>
         q.findRecord(locationSignature)
       );
-      let measurements: Measurement[] = await store.query((q) =>
+    } catch {
+      // Fail silently
+    }
+    try {
+      // Fetch data from the API
+      let transform: Transform[] = await remoteStore.pull((q) =>
+        q.findRelatedRecords(locationSignature, 'measurements')
+      );
+
+      // Remove old data
+      let cachedResults = store.cache.query((q) =>
         q.findRelatedRecords(locationSignature, 'measurements')
       );
 
-      if (location && measurements) {
-        this.location = location;
-        this.measurements = measurements;
+      // Remove previous relationships and records
+      await store.update((t) => {
+        return cachedResults.reduce((accumulator, result) => {
+          let measurementSignature = { type: 'measurement', id: result.id };
+          let deleteRelationship = t.removeFromRelatedRecords(
+            locationSignature,
+            'measurements',
+            measurementSignature
+          );
+          let deleteRecord = t.removeRecord(
+            measurementSignature
+          );
+          return [...accumulator, deleteRelationship, deleteRecord];
+        }, [] as Transform[]);
+      });
```

Comment:
I think we should only be running this block when we read new measurements from the API successfully but I'm not sure that is checked anywhere?

Reply:
I think we should keep both API calls in separate `try`s. In case the first call fails, we can still try the second one and perhaps we have from cache the location so we can still serve the page with fresh data. If we put both call in the same `try`, we make the second call dependent on the first call.

It's still valid to say that we can allow the first call to fail silently because perhaps the location was found in the cache. If it wasn't there's a computed property that would trigger an error so we don't need to stop the flow here. 

Also, we only clean the cache if the API call for the measurements was successful. Any error on the store `pull` would throw an exception that would abort the rest of the instructions in the `try` block.

Reply:
I still don't understand this. Isn't [this line](https://github.com/simplabs/breethe-client/pull/97/files#diff-c1464c0d7c484090f2e39089a8ba4814R204) where data is being loaded from the API but we're already deleting records from `IndexedDB` [here](https://github.com/simplabs/breethe-client/pull/97/files#diff-c1464c0d7c484090f2e39089a8ba4814R188)? We should find some time to pair on this I think.

Reply:
No, [this line](https://github.com/simplabs/breethe-client/pull/97/files#diff-c1464c0d7c484090f2e39089a8ba4814R204) syncs the loaded records with the store. We pull the [here](https://github.com/simplabs/breethe-client/pull/97/files#diff-c1464c0d7c484090f2e39089a8ba4814R178). But I agree we should pair on this to make everything clearer, I'll ping you.